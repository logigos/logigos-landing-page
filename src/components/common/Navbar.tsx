import React from 'react'
import styled from 'styled-components'
import Button from './Button'
import R from '../resources/R'
import media from 'styled-media-query'

const NavRoot = styled.nav`
    display: flex;
    align-items: center;
    justify-content: flex-end;
    position: absolute;
    top: 4.8rem;
    right: 4.8rem;
    z-index: 1;
`

const ButtonCustom = styled(Button)`
    ${media.lessThan('small')`
        font-size: 1.4rem;
        width: 5.7rem;
        height: 2.2rem;
    `}
`

const Navbar = () => {
    return (
        <NavRoot>
            <ButtonCustom
                bg="transparent"
                hbg={R.colors.black_l_60}
                abg={R.colors.black_l_90}
                height="4.7rem"
                width="15rem"
                font_size="2.4rem"
                margin="auto 1.2rem"
            >
                เข้าสู่ระบบ
            </ButtonCustom>
            <ButtonCustom
                bg="transparent"
                hbg="white"
                abg={R.colors.black_l_90}
                height="4.7rem"
                width="18rem"
                font_size="2.4rem"
                border={`solid .3rem white`}
                aborder={`solid .3rem ${R.colors.black_l_90}`}
                hcolor={R.colors.black_l_40}
                margin="auto 1.2rem"
            >
                สมัครสมาชิก
            </ButtonCustom>
        </NavRoot>
    )
}

export default Navbar
