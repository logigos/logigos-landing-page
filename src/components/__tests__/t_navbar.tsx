import React from 'react'
import renderer from 'react-test-renderer'

import Navbar from '../common/Navbar'

describe('Navbar', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Navbar />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
