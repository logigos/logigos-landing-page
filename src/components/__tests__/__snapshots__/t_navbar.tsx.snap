// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="sc-AxirZ iFWzZN"
>
  <button
    className="sc-AxjAm gvAePj"
    height="4.7rem"
    width="15rem"
  >
    เข้าสู่ระบบ
  </button>
  <button
    className="sc-AxjAm esTlVd"
    height="4.7rem"
    width="18rem"
  >
    สมัครสมาชิก
  </button>
</nav>
`;
